docs/proposals/version_documentation.md

```diff
@@ -0,0 +1,76 @@
+---
+title: 'Documentation Versioning'
+linkTitle: 'Documentation Versioning'
+---
+
+- Author: [Uchechukwu Obasi](https://github.com/thisisobate)
+- Date: May 2020
+- Status: proposal
+
+## Problem
+
+The Thanos website contains the docs area which is built by rendering the markdown files from the tip of master of the Thanos repository. This frequently causes confusion for users, as breaking changes are often merged into master that conflicts with the APIs of previous releases.
+
+In this proposal, we want to solve this. In particular, we want to:
+
+1. Build a Version Picker that serves as a tool which will aid easy access to other versions of the documentation. This picker will Include links to change version (the version must be in the URL).
+2. A solid and robust Documentation structure.
+3. Version specific pages of the documentation
```

Comment:
What do you mean here? I think the granularity would be for all docs, not per single page? :thinking:  Whatever that means (:

Reply:
Excatly @bwplotka The ultimate idea is to have all the docs of the Thanos project versioned.

I'm actively getting alot of insights on how we could build a plugin to automate the versioning process. Surprisingly, I wasn't able to find any similar Hugo versioning plugin which presents an opportunity for us to build a useful solution that will be beneficial not only to Thanos, but also to the entire Hugo community as well 💯